### DISCUSSION TITLE
Pie chart not taking the data

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2215] [Help] [answered]
Hi,

while playing around with the widget/ chart part of filament I tried to have a new piechart widget using the following getData():
```
  protected function getData(): array
    {
        // Load all active third parties as $data
        $data = ThirdParty::where('is_active', true)->get();

        // Load all types of third parties into array $types
        $types = [];
        foreach ($data as $thirdParty) {
            $types[] = $thirdParty->type;
        }

        return [
            'datasets' => [
                [
                    'label' => 'Test label',
                    // Number of instances re types of third parties
                    'data' => count(array_unique($types)),
                ],
            ],
            // Array with all unique types of third parties
                'labels' => array_unique($types),
        ];
    }
```

All I receive is this:
![grafik](https://user-images.githubusercontent.com/67121752/164983673-ea88a61d-1a6a-42ec-ba9a-4f5d3937c3af.png)

While the labels seem to work fine (it's 3 seeded third parties: 2 suppliers, 1 client), the data is not taken.
Does anyone have a suggestion what is wrong with this setup? Is is count(array_unique($types)) as this might not provide the right form of array needed?

Thanks!

**Comment by danharrin** [comment, marked as answer]
what array is returned from `getData()` ?

**Reply by vv-ac** [reply]
Thank you for getting back. 
As I am learning Laravel/PHP besides having a full-time job, it might take some time for me to answer - but here I am again ;-).

How would you echothe array values in this case? Putting an echo in the return does not work.

**Reply by danharrin** [reply]
```php
        return dd([
            'datasets' => [
                [
                    'label' => 'Test label',
                    // Number of instances re types of third parties
                    'data' => count(array_unique($types)),
                ],
            ],
            // Array with all unique types of third parties
                'labels' => array_unique($types),
        ]);
```

**Reply by vv-ac** [reply]
Thank you! Here is the output:
```
^ array:2 [▼
  "datasets" => array:1 [▼
    0 => array:2 [▼
      "label" => "Test label"
      "data" => 2
    ]
  ]
  "labels" => array:2 [▼
    0 => "Supplier"
    1 => "Client"
  ]
]
```

**Reply by vv-ac** [reply]
So, it seems like it is not counting the number the different types of "types" occur in the array but simply the number of types itself - which then is obviouly not fit to the two labels.
Will work on it!

**Reply by danharrin** [reply]
It needs to fit this format:

<img width="338" alt="Screenshot 2022-04-27 at 09 51 59" src="https://user-images.githubusercontent.com/41773797/165480538-ed2c10f5-0a7e-44af-ba6a-7ad7d5389938.png">

**Reply by vv-ac** [reply]
Thanks again for helping out, Dan!
This did the trick:
```
        $data = [];
        foreach ($types as $type) {
            if (isset($data[$type])) {
                $data[$type]++;
            } else {
                $data[$type] = 1;
            }
        }
```

```
        return [
            'datasets' => [
                [
                    'label' => 'Test label',
                    // Number of instances re types of third parties
                    'data' => array_values($data),
                ],
            ],
            // Array with all unique types of third parties
                'labels' => array_unique($types),
        ];
```

**Comment by vv-ac** [comment]
Trying to reduce the size of the widget I used in the widget class:
```
    protected int | string | array $columnSpan = 2;
```

That seems not to be working, though it should use the following:
```
match ($this->getColumnSpan()) {
        2 => 'col-span-2',
        3 => 'col-span-3',
        4 => 'col-span-4',
        5 => 'col-span-5',
        6 => 'col-span-6',
        7 => 'col-span-7',
        8 => 'col-span-8',
        9 => 'col-span-9',
        10 => 'col-span-10',
        11 => 'col-span-11',
        12 => 'col-span-12',
        'full' => 'col-span-full',
        1 => 'col-span-1',
    },
```

'full' is working, whereas no number is taken, it always switches to full then.
Am I using the $columnSpan in the wrong way?
